arch/src/lib.rs

```diff
@@ -96,6 +101,46 @@ fn pagesize() -> usize {
     unsafe { libc::sysconf(libc::_SC_PAGESIZE) as usize }
 }
 
+#[derive(Clone, Default)]
+pub struct NumaNode {
+    pub memory_regions: Vec<Arc<GuestRegionMmap>>,
+    pub hotplug_regions: Vec<Arc<GuestRegionMmap>>,
+    pub cpus: Vec<u8>,
+    pub distances: BTreeMap<u32, u8>,
+    pub memory_zones: Vec<String>,
+    #[cfg(target_arch = "x86_64")]
+    pub sgx_epc_sections: Vec<SgxEpcSection>,
+}
+
+impl NumaNode {
```

Comment:
What's the benefit of having all these accessors if all the fields are pub?

Reply:
To be honest I don't know....I didn't think too much when moving this part and just moved the original code as it is.

Maybe @sboeuf is more experienced to answer this question?

Reply:
Check if the code is doing extensive use of directly accessing the fields. If not, then try to make sure it relies on accessors only.

Reply:
I think you can remove the accessors if all the fields are pub.

Reply:
Hi @rbradford. Yes and done. Such change is reflected in https://github.com/cloud-hypervisor/cloud-hypervisor/pull/2950/commits/fdd0a6c469b638ee5c79ee059a71bb9d2f1c578c